echo.go

```diff
@@ -93,6 +100,9 @@ type (
 		Logger           Logger
 		IPExtractor      IPExtractor
 		ListenerNetwork  string
+
+		// OnAddRouteHandler is called when Echo adds new route to specific host router.
+		OnAddRouteHandler func(host string, route Route, handler HandlerFunc, middleware []MiddlewareFunc)
```

Comment:
Should the OnAddRouteHandler be able to return an error, that might stop adding the route?
(just a question whether your thought about that)

Reply:
I did not add error here because this handler is called after route has been added, because during adding, in lower level, the route is created, which we pass into handler.